### PR TITLE
Faster dsm ci

### PR DESCRIPTION
**Pull request by tseho**
**Description (for Contributor and Core Developer)**

Same as https://github.com/akeneo/pim-enterprise-dev/pull/9996
**But with a difference**, due to the build being either CE only or CE in EE, the path is not the same in those 2 cases.
There is a new parameter `<< parameters.path_to_dsm >>`.

**CE Only without cache:**
---

Screenshot_2020-12-04_17-32-23
Screenshot_2020-12-04_17-32-46

**CE Only with cache:**
---

Screenshot_2020-12-04_17-44-11
Screenshot_2020-12-04_17-44-27

**CE in EE without cache:**
---

Screenshot_2020-12-04_17-33-13
Screenshot_2020-12-04_17-33-25

**CE in EE with cache:**
---

Screenshot_2020-12-04_17-52-04
Screenshot_2020-12-04_17-52-18

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added legacy Behats               | Todo
| Added acceptance tests            | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -112,6 +112,56 @@ jobs:
       - run:
           name: Build css
           command: make css
+      - when:
+            condition: << parameters.is_ee_built >>
```
Comment (ahocquard) [inline .circleci/config.yml:116]: I would avoid copy/past and specify << parameters.path_to_dsm >> in the workflow I think

Comment (tseho) [reply .circleci/config.yml:116]: I thought about it but I'm not confident enough with circleci variables.
I can try.

Comment (ahocquard) [reply .circleci/config.yml:116]: I can show you how to do if you want.

**Review by ValentinMumble — APPROVED** [review]
👏

**Review by mr-prud — APPROVED** [review]

**Review by BitOne — APPROVED** [review]

**Review by juliensnz — APPROVED** [review]

File: .circleci/config.yml [hunk 2]
```diff
@@ -563,6 +588,7 @@ workflows:
                 requires:
                     - ready_to_build?
           - build_dev:
+                path_to_dsm: vendor/akeneo/pim-community-dev/akeneo-design-system
```
Comment (ahocquard) [inline .circleci/config.yml:591]: where is the one for CE?

Comment (ahocquard) [reply .circleci/config.yml:591]: Because it's only triggered by nightly builds, you didn't put it maybe?

Comment (tseho) [reply .circleci/config.yml:591]: there: https://github.com/akeneo/pim-community-dev/pull/13441/files#diff-78a8a19706dbd2a4425dd72bdab0502ed7a2cef16365ab7030a5a0588927bf47R64

**Review by ahocquard — APPROVED** [review]

**Pull request merged**